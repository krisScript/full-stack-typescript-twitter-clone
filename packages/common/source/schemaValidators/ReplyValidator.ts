/* eslint-disable import/no-duplicates */
import * as yup from 'yup';

interface ReplyValidatorType {
  text: string;
}
const ReplyValidator = yup.object<ReplyValidatorType>().shape({
  text: yup
    .string()
    .trim()
    .min(1)
    .max(500),
});
export default ReplyValidator;
