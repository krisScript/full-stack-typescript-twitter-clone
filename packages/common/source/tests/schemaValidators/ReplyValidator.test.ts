import ReplyValidator from '@schemaValidators/ReplyValidator';

describe('ReplyValidator', (): void => {
  const text =
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique vel alias, amet corporis modi corrupti.';
  it(`should validate successfully`, async (): Promise<void> => {
    const reply = {
      text,
    };
    await expect(
      ReplyValidator.validate(reply, { abortEarly: false }),
    ).resolves.toBe(reply);
  });
  it(`should throw an error`, async (): Promise<void> => {
    const reply = {
      text: '',
    };
    await expect(
      ReplyValidator.validate(reply, { abortEarly: false }),
    ).rejects.toMatchSnapshot();
  });
});
