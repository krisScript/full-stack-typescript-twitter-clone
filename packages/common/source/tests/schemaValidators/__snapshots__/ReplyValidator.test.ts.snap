// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReplyValidator should throw an error 1`] = `[ValidationError: text must be at least 1 characters]`;
